File: schemas/compliance-grc/convex/control_requirements.ts
// control_requirements: Maps controls to framework requirements (many-to-many).
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const controlRequirements = defineTable({
  controlId: v.id("controls"),
  requirementId: v.id("framework_requirements"),
  notes: v.optional(v.string()),
})
  .index("by_control_id_and_requirement_id", ["controlId", "requirementId"]);

File: schemas/compliance-grc/spacetimedb/control_requirements.rs
// control_requirements: Maps controls to framework requirements (many-to-many).
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = control_requirements, public)]
pub struct ControlRequirement {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub control_id: String, // UUID — FK → controls.id (cascade delete)
    // Composite unique: (control_id, requirement_id)

    #[index(btree)]
    pub requirement_id: String, // UUID — FK → framework_requirements.id (cascade delete)

    pub notes: Option<String>,

    pub created_at: Timestamp,
}
